rust-semantics/expression/struct.md

```diff
@@ -5,8 +5,89 @@ module RUST-EXPRESSION-STRUCT
     imports private RUST-CONVERSIONS-SYNTAX
     imports private RUST-ERROR-SYNTAX
     imports private RUST-EXECUTION-CONFIGURATION
+    imports private RUST-PREPROCESSING-CONFIGURATION
     imports private RUST-REPRESENTATION
 
+    // For freezing heating operations, when leaving a struct alone in the topcell, make it a ptrValue
+    rule <k> struct(P, Fields) => ptrValue(null, struct(P, Fields)) ... </k>
+
+    // KItems for breaking down struct expressions into the adequate format struct(P, F)
+    syntax KItem ::= fromStructExpressionWithLiteralsBuildFieldsMap(TypePath, LiteralExpressionList, List, Map)
+                   | fromStructExpressionWithAssignmentsBuildFieldsMap(TypePath, StructFieldAssignments, Map)
+
+
+    // From Struct Expression to struct(P, F). Case 1, field names are not given:
+    rule <k> 
+            I:TypePath { L:LiteralExpressionList } => fromStructExpressionWithLiteralsBuildFieldsMap(I, L, VL, .Map) 
+            ... 
+         </k>
+        <struct-path> I </struct-path>
+        <variable-list> VL </variable-list>
+        
+    rule <k> fromStructExpressionWithLiteralsBuildFieldsMap(I:TypePath, (E:LiteralExpression, RL):LiteralExpressionList, FieldNameList:List, FieldsMap:Map)
+                => E ~> fromStructExpressionWithLiteralsBuildFieldsMap(I, RL, FieldNameList, FieldsMap) ... 
+         </k>
+
+    rule 
+        <k> 
+            ptrValue(_, V:Value):PtrValue ~> 
+            fromStructExpressionWithLiteralsBuildFieldsMap(
+                    I:TypePath, 
+                    RL:LiteralExpressionList, 
+                    (FieldNameList):List ListItem(FieldName), 
+                    FieldsMap:Map)
+            => fromStructExpressionWithLiteralsBuildFieldsMap(I, RL, FieldNameList, FieldsMap (FieldName |-> NVI):Map )  
```

Comment:
I think it would be better to add a requires clause checking that FieldName is not in FieldsMap.